src/main/java/Nocturne.java

```diff
@@ -0,0 +1,94 @@
+                    tasks[listIndex - 1].isDone = true;
+                    System.out.println("  " + tasks[listIndex - 1]);
+                    break;
+                }
+                case "unmark": {
+                    if (commandCheck.length < 2 | commandCheck.length > taskCount + 1) {
+                        throw new NocturneException("Your list is either empty or your brain is.");
+                    }
+                    int listIndex = Integer.parseInt(commandCheck[1]);
+                    System.out.println("Do not neglect your duties. I have marked this task as unfinished:");
+                    tasks[listIndex - 1].isDone = false;
+                    System.out.println("  " + tasks[listIndex - 1]);
+                    break;
+                }
+                case "deadline":
+                    String[] deadlineSeparated = input.split("/");
+                    if (deadlineSeparated.length != 3) {
+                        throw new NocturneException("Take your / back, and only put 2!");
+                    }
+                    Deadline trueDeadline;
+                    String deadlineName = deadlineSeparated[0].substring(9);
+                    String by = deadlineSeparated[1].substring(3);
+                    trueDeadline = new Deadline(deadlineName, by);
+                    tasks[taskCount] = trueDeadline;
+                    taskCount++;
+                    System.out.println("A deadline I see. I have added it:");
+                    System.out.println("  " + trueDeadline);
+                    break;
+
+                case "todo":
+                    String todoName = input.substring(5);
+                    Todo trueTodo = new Todo(todoName);
+                    tasks[taskCount] = trueTodo;
+                    taskCount++;
+                    System.out.println("A Todo task I see. I have added it:");
+                    System.out.println("  " + trueTodo);
+                    break;
+
+                case "event":
+                    String[] eventSeparated = input.split("/");
+                    if (eventSeparated.length != 3) {
+                        throw new NocturneException("Take your / back, and only put 2!");
+                    }
+                    Event trueEvent;
+                    String eventName = eventSeparated[0].substring(6);
+                    String from = eventSeparated[1].trim();
+                    from = from.substring(5);
+                    String to = eventSeparated[2].substring(3);
+                    trueEvent = new Event(eventName, from, to);
+                    tasks[taskCount] = trueEvent;
+                    taskCount++;
+                    System.out.println("An event I see. I have added it:");
+                    System.out.println("  " + trueEvent);
+                    break;
+
+                default:
+                    throw new NocturneException("Your command is invalid. Try again.");
+            }
```

Comment:
switch statement should be in the following form:

```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

Reply:
additionally, might want to consider moving & refactoring the different operations to different methods (e.g. listTasks(),  markTask()) so that everything is not under the main method! try to avoid long methods :]